src/Symfony/Component/Security/Http/Authenticator/AccessTokenAuthenticator.php

```diff
@@ -58,10 +58,11 @@ public function authenticate(Request $request): Passport
         if (!$accessToken) {
             throw new BadCredentialsException('Invalid credentials.');
         }
-        $userIdentifier = $this->accessTokenHandler->getUserIdentifierFrom($accessToken);
+
+        $userIdentifier = $this->accessTokenHandler->getUserFrom($accessToken);
 
         return new SelfValidatingPassport(
-            new UserBadge($userIdentifier, $this->userProvider->loadUserByIdentifier(...))
+            $userIdentifier instanceof UserBadge ? $userIdentifier : new UserBadge($userIdentifier, $this->userProvider->loadUserByIdentifier(...))
```

Comment:
Is an explicit check and throw needed here?

```suggestion
            $userIdentifier instanceof UserBadge ? $userIdentifier : new UserBadge($userIdentifier, ($this->userProvider ?? throw new \Symfony\Component\Security\Core\Exception\RuntimeException('No UserProvider given'))->loadUserByIdentifier(...))
```

Reply:
Yes for the check, falling back to null

```suggestion
            $userIdentifier instanceof UserBadge ? $userIdentifier : new UserBadge($userIdentifier, $this->userProvider ? $this->userProvider->loadUserByIdentifier(...) : null)
```